src/functionalTest/java/uk/gov/hmcts/reform/waworkflowapi/controllers/EvaluateDmnTest.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.waworkflowapi.controllers;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.hmcts.reform.waworkflowapi.SpringBootFunctionalBaseTest;
+import uk.gov.hmcts.reform.waworkflowapi.controllers.startworkflow.ServiceDetails;
+import uk.gov.hmcts.reform.waworkflowapi.external.taskservice.DmnValue;
+import uk.gov.hmcts.reform.waworkflowapi.external.taskservice.EvaluateDmnRequest;
+import uk.gov.hmcts.reform.waworkflowapi.utils.AuthorizationHeadersProvider;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static net.serenitybdd.rest.SerenityRest.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.waworkflowapi.config.ServiceTokenGeneratorConfiguration.SERVICE_AUTHORIZATION;
+
+public class EvaluateDmnTest extends SpringBootFunctionalBaseTest {
+
+
+    @Value("${targets.instance}")
+    private String testUrl;
+
+    @Autowired
+    private AuthorizationHeadersProvider authorizationHeadersProvider;
+
+    private String caseId;
+
+    private String serviceAuthorizationToken;
+
+    @Before
+    public void setUp() {
+        caseId = UUID.randomUUID().toString();
+        serviceAuthorizationToken =
+            authorizationHeadersProvider
+                .getAuthorizationHeaders()
+                .getValue(SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    public void should_not_allow_requests_without_valid_service_authorisation_and_return_403_response_code() {
```

Comment:
I know some of this is from what Chris initially did.  Are there some more negative tests covered elsewhere?  E.g. if you provide an invalid `id`.

Reply:
I have added tests in for both now which handle these situations